Jenkinsfile

```diff
@@ -88,61 +73,75 @@ pipeline {
 				}
 			}
 		}
-		// This stage creates a new 'release_current' database from the freshly updated 'slice_current' database.
-		// This will be the primary database used throughout release from here.
-		stage('Post: Create release_current from slice_current'){
+		// This stage runs StableIdentifier QA, which checks all stable ids in the database for valid formats.
+		// Currently this module comes from data-release-pipeline@feature/post-step-stid-history, but in
+		// the future will be moved to a QA repository, specific to release.
+		stage('Post: Build and run StableIdentifier QA') {
 			steps{
 				script{
-					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]) {
-						sh "mysql -u$user -p$pass -e \'drop database if exists ${env.RELEASE_CURRENT}; create database ${env.RELEASE_CURRENT}\'"
-						sh "mysqldump --opt -u$user -p$pass ${env.SLICE_CURRENT} | mysql -u$user -p$pass ${env.RELEASE_CURRENT}"
+					// Clone data-release-pipeline and checkout specific branch
+					utils.cloneOrUpdateLocalRepo("data-release-pipeline")
+					dir("data-release-pipeline") {
+				        	sh "git checkout feature/post-step-tests-stid-history"
+				        
+						// Build and run QA jar file
+						dir("ortho-stable-id-history") {
+							utils.buildJarFile()
+							withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]) {
+								sh "java -jar target/OrthoStableIdHistory-*-jar-with-dependencies.jar $ConfigFile"
+							}
+						}
 					}
 				}
 			}
 		}
-		// QA for ensuring StableIdentifier instances are proper.
-		stage('Post: StableIdentifier QA'){
+		// This stage creates a new 'release_previous' database from the 'release_current' database,
+		// and takes the recently updated 'slice_current' database and creates a new 'release_current' one.
+		stage('Post: Rotate release DBs'){
 			steps{
 				script{
-					dir("data-release-pipeline/ortho-stable-id-history"){
-						withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]) {
-							sh "java -jar target/OrthoStableIdHistory-*-jar-with-dependencies.jar $ConfigFile"
-						}
+					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]) {
+						
+						// Take existing 'release_current' DB and replace 'release_previous' with it.
+						// TODO this should come from S3 final DB resting place.
+						def previousReleaseVersion = utils.getPreviousReleaseVersion()
+						def previousReleaseCurrentFinalFilename = "${env.RELEASE_CURRENT_DB}_${previousReleaseVersion}_final.dump"
```

Comment:
I was initially a little confused which database this variable is referring to since it is described as "previous" and "release current" in the variable name.  Maybe this can be named something like "releaseCurrentDBToBeReplacedDumpName"?

Reply:
Sure, that makes sense -- I struggled coming up with a name that was both explicit and not unwieldy (I failed)

Reply:
Done